library/std/src/sys/locks/rwlock/queue.rs

```diff
@@ -343,9 +347,10 @@ impl RwLock {
                 // pointer to the next node in the queue. Otherwise set it to
                 // the lock count if the state is read-locked or to zero if it
                 // is write-locked.
-                node.next.0 = AtomicPtr::new(state.mask(MASK).cast());
-                node.prev = AtomicLink::new(None);
-                let mut next = ptr::from_ref(&node)
+                // These writes can't race.
```

Comment:
I'm not actually sure why we know that they can't race. After all, as the Miri error shows, an alias can clearly still exist when these writes happen. This assumption is not new; we've been using non-atomic writes here already before this PR. Previously this was safe code only because rustc did not realize that the `NonNull::from(&node)` creates a borrow that could last into the next loop iteration.